src/thor/route_action.cc

```diff
@@ -211,6 +354,13 @@ void thor_worker_t::route(Api& request) {
   auto& options = *request.mutable_options();
   adjust_scores(options);
   controller = AttributesController(options);
+
+  if (!request.options().cost_factor_lines().empty()) {
+    parse_costing(request);
```

Comment:
we're parsing costing twice now in this path. looks like you could just move the outer one before this if?

Reply:
missed that completely, thanks!

Reply:
I'd leave a small comment here too. that looks like smth I'd try to change on a whim sometime in the future;)